xarray/core/variable.py

```diff
@@ -42,7 +42,7 @@ def as_variable(obj, name=None, copy=False):
         obj = obj.copy(deep=False)
     elif hasattr(obj, 'dims') and (hasattr(obj, 'data') or
                                    hasattr(obj, 'values')):
-        obj = Variable(obj.dims, getattr(obj, 'data', obj.values),
+        obj = Variable(obj.dims, getattr(obj, 'data', getattr(obj, 'values')),
```

Comment:
Actually I was confused when I switched this. It does literally the exact
same thing - I will revert. My concern was that this might prematurely
evaluate the data if it's a dask array.
On Tue, Aug 16, 2016 at 9:16 PM Maximilian Roos <notifications@github.com>
wrote:

> In xarray/core/variable.py
> <https://github.com/pydata/xarray/pull/971#discussion_r75060357>:
>
> > @@ -42,7 +42,7 @@ def as_variable(obj, name=None, copy=False):
> >          obj = obj.copy(deep=False)
> >      elif hasattr(obj, 'dims') and (hasattr(obj, 'data') or
> >                                     hasattr(obj, 'values')):
> > -        obj = Variable(obj.dims, getattr(obj, 'data', obj.values),
> > +        obj = Variable(obj.dims, getattr(obj, 'data', getattr(obj, 'values')),
>
> Out of interest, why is this better?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pydata/xarray/pull/971/files/25bebc8c427f5947545ba3de101ca02108754037#r75060357>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABKS1s27hTk1CurNGfup--qbCecMSdZaks5qgosBgaJpZM4JmDcq>
> .
>